src/tools/decomp.py

```diff
@@ -53,47 +99,284 @@
+	j=[]
+	for i in xrange(0,MnN):
+	    if (indMax==0 and indMin==1) or (indMax==0 and indMin==2):
+		if nX[i]>(SymFactor-percTol*SymFactor)*nY[i] and nX[i]<(SymFactor+percTol*SymFactor)*nY[i]:
+		    j.append(i)
+	    elif (indMax==1 and indMin==0) or (indMax==1 and indMin==2):
+		if nY[i]>(SymFactor-percTol*SymFactor)*nX[i] and nY[i]<(SymFactor+percTol*SymFactor)*nX[i]:
+		    j.append(i)            
+	    elif (indMax==2 and indMin==0) or (indMax==2 and indMin==1):
+		if nZ[i]>(SymFactor-percTol*SymFactor)*nX[i] and nZ[i]<(SymFactor+percTol*SymFactor)*nX[i]:
+		    j.append(i)            
+	print "HDD message: There are a couple of total number of processors that satisfy your requirements: ",[(j[i]+1)*CpN for i in xrange(len(j))]
+	return [(j[i]+1)*CpN for i in xrange(len(j))]
+
+# WARNING! This is a new function to find values for the new neighborList DataStruct for Hom-Sys
+def neiListHom(node_grid,box,rc,skin):
+    # data structure Initialization
+    print "HDD message: Current homogeneous NodeGrid (X,Y,Z) -> powered by HeSpaDDA",node_grid
+    rc_skin=rc+skin	
+    neiListxin=[]
+    neiListyin=[]
+    neiListzin=[]
+    cursor=[box[0],box[1],box[2]]
+    #define Neighbor vectors
+    neiListx=[0]*(node_grid[0]+1)
+    neiListy=[0]*(node_grid[1]+1)
+    neiListz=[0]*(node_grid[2]+1)
+    # It makes use of reDistCellsHom and adaptNeiList form loadbal to find homogenous DD  	    
+    neiListxin=reDistCellsHom(node_grid[0],cursor[0],rc_skin)
+    neiListx=adaptNeiList(neiListxin)
+    neiListyin=reDistCellsHom(node_grid[1],cursor[1],rc_skin)
+    neiListy=adaptNeiList(neiListyin)
+    neiListzin=reDistCellsHom(node_grid[2],cursor[2],rc_skin)
+    neiListz=adaptNeiList(neiListzin)
+    return map(int,neiListx),map(int,neiListy),map(int,neiListz)
+
+# WARNING! This is a new function to find values for the new neighborList DataStruct for inHom-Sys
+def neiListAdress(node_grid, cell_grid,rc,skin,eh_size,adrCenter,ratioMS,idealGasFlag=True,sphereAdr=False,slabMSDims=[1,0,0]):
+    # dataStructure Initialization
+    print "HDD message: Current heterogeneous NodeGrid (X,Y,Z) -> powered by HeSpaDDA:",node_grid
+    rc_skin=rc+skin
+    #define Neighbor vectors
+    neiListx=[0]*(node_grid[0]+1)
+    neiListy=[0]*(node_grid[1]+1)
+    neiListz=[0]*(node_grid[2]+1)
+    # Check adress regions
+    adrCenter=[adrCenter[0],adrCenter[1],adrCenter[2]]
+    # Midle point DD
+    cursor=[adrCenter[0]*2,adrCenter[1]*2,adrCenter[2]*2]
+    #cg_sizeR=adrCenter[0]+eh_size 	# for further devs
+    #cg_sizeL=adrCenter[0]-eh_size	# for further devs
+    cellsX=round(cursor[0]/ rc_skin-0.5)#round(cg_sizeL/rc_skin-0.5)+round((cursor[0]-cg_sizeR)/ rc_skin-0.5)+round((cg_sizeR-cg_sizeL)/rc_skin-0.5) # old implementation rounded # Cells too much!
+    cellsY=round(cursor[1] / rc_skin-0.5)
+    cellsZ=round(cursor[2] / rc_skin-0.5)
+    print "HDD message: Current heterogeneous CellGrid (X,Y,Z) -> powered by HeSpaDDA:",cellsX,cellsY,cellsZ
```

Comment:
I would really prefer to not to see `powered by HeSpaDDA` multiple times.

Reply:
OK1